### PR TITLE
Add note about in-place updating in the docs

### PR DESCRIPTION
**Pull request by MatteoRobbiati**
Note in the docs as decided in #1037.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: doc/source/getting-started/backends.rst [hunk 1]
```diff
@@ -50,6 +50,13 @@ be controlled using the ``QIBO_LOG_LEVEL`` environment variable. This can be set
 to 3 to hide info messages or 4 to hide both info and warning messages. The
 default value is 1 allowing all messages to appear.
 
+.. note::
+  The `qibojit` backend implements in-place updating of states and unitaries. This
+  implies a target initial state is modified at runtime, in order to save
+  memory. In the case of algorithms in which the initial state is used several times, or
+  it is necessary to be maintain it over time (such as in quantum machine learning applications)
+  it is advisable to save a copy of the states.
```
Comment (andrea-pasquale) [inline doc/source/getting-started/backends.rst:58]:
```suggestion
  The `qibojit` backend implements in-place updates. This
  implies that the initial state is modified at runtime without performing any copies to save memory.
  For algorithms that require access to the initial state after its modification, such as quantum machine learning applications,
  it is suggested to perform a copy of the state explicitly.
```

**Review by andrea-pasquale — COMMENTED** [review]
Thanks @MatteoRobbiati.
I reformulated the paragraph quickly.
Let me know if you agree.

**Comment by MatteoRobbiati** [comment]
Thanks @andrea-pasquale.

**Review by andrea-pasquale — APPROVED** [review]

File: doc/source/getting-started/backends.rst [hunk 2]
```diff
@@ -50,6 +50,12 @@ be controlled using the ``QIBO_LOG_LEVEL`` environment variable. This can be set
 to 3 to hide info messages or 4 to hide both info and warning messages. The
 default value is 1 allowing all messages to appear.
 
+.. note::
+  The `qibojit` backend implements in-place updates. This
+  implies that the initial state is modified at runtime without performing any copies to save memory.
+  For algorithms that require access to the initial state after its modification, such as quantum machine learning applications,
+  it is suggested to perform a copy of the state explicitly.
```
Comment (scarrazza) [inline doc/source/getting-started/backends.rst:57]:
```suggestion
  it is suggested to perform a copy of the state explicitly if really needed by the algorithm.
```

**Pull request merged**